libraries/node-core-library/src/test/Async.test.ts

```diff
@@ -313,4 +313,128 @@ describe(Async.name, () => {
+      const expectedResult: string = 'RESULT';
+      const result: string = await Async.runWithRetriesAsync(async () => expectedResult, 0);
+      expect(result).toEqual(expectedResult);
+    });
+
+    it('Correctly handles a sync function that throws and does not allow retries', async () => {
+      await expect(
+        async () =>
+          await Async.runWithRetriesAsync(() => {
+            throw new Error('error');
+          }, 0)
+      ).rejects.toThrowErrorMatchingSnapshot();
+    });
+
+    it('Correctly handles an async function that throws and does not allow retries', async () => {
+      await expect(
+        async () =>
+          await Async.runWithRetriesAsync(async () => {
+            throw new Error('error');
+          }, 0)
+      ).rejects.toThrowErrorMatchingSnapshot();
+    });
+
+    it('Correctly handles a sync function that always throws and allows several retries', async () => {
+      await expect(
+        async () =>
+          await Async.runWithRetriesAsync(() => {
+            throw new Error('error');
+          }, 5)
+      ).rejects.toThrowErrorMatchingSnapshot();
+    });
+
+    it('Correctly handles an async function that always throws and allows several retries', async () => {
+      await expect(
+        async () =>
+          await Async.runWithRetriesAsync(async () => {
+            throw new Error('error');
+          }, 5)
+      ).rejects.toThrowErrorMatchingSnapshot();
+    });
+
+    it('Correctly handles a sync function that throws once and then succeeds', async () => {
+      const expectedResult: string = 'RESULT';
+      let callCount: number = 0;
+      const result: string = await Async.runWithRetriesAsync(() => {
+        if (callCount++ === 0) {
+          throw new Error('error');
+        } else {
+          return expectedResult;
+        }
+      }, 1);
+      expect(result).toEqual(expectedResult);
+    });
+
+    it('Correctly handles an async function that throws once and then succeeds', async () => {
+      const expectedResult: string = 'RESULT';
+      let callCount: number = 0;
+      const result: string = await Async.runWithRetriesAsync(() => {
+        if (callCount++ === 0) {
+          throw new Error('error');
+        } else {
+          return expectedResult;
+        }
+      }, 1);
+      expect(result).toEqual(expectedResult);
+    });
+
+    it('Correctly handles a sync function that throws once and then succeeds with a timeout', async () => {
+      const expectedResult: string = 'RESULT';
+      let callCount: number = 0;
+      const sleepSpy: jest.SpyInstance = jest
+        .spyOn(Async, 'sleep')
+        .mockImplementation(() => Promise.resolve());
+
+      const resultPromise: Promise<string> = Async.runWithRetriesAsync(
+        () => {
+          if (callCount++ === 0) {
+            throw new Error('error');
+          } else {
+            return expectedResult;
+          }
+        },
+        1,
+        5
```

Comment:
The API parameters are not very readable:
```ts
const resultPromise: Promise<string> = Async.runWithRetriesAsync(() => {
    // since there may be many lines of code here,
    // better to put this last
  },
  1,  // what is 1?
  5   // what is 5? what are the units? no idea!
);
```
compare with
```ts
const resultPromise: Promise<string> = Async.runWithRetriesAsync(
  { maxRetries: 1, retryDelayMs: 5 },
  () => {
    // since there may be many lines of code here,
    // better to put this last
  }
);
```
or maybe even better:
```ts
const resultPromise: Promise<string> = Async.runWithRetriesAsync({
  maxRetries: 1,
  retryDelayMs: 5,
  action: () => {
    // since there may be many lines of code here,
    // better to put this last
  },
});
```
it also allows the possibility for more options in the future, e.g. exponentially increasing delays